### PR TITLE
lua-cli improvements

### PR DESCRIPTION
**Pull request by daurnimator**

### CODE REVIEW

File: tests/manual-tests/lua-cli.js [hunk 1]
```diff
@@ -24,10 +26,37 @@ for (;;) {
     if (input.length === 0)
         continue;
 
-    let buffer = lua.to_luastring(input);
-    if (lauxlib.luaL_loadbuffer(L, buffer, buffer.length, lua.to_luastring("=stdin"))
-        || lapi.lua_pcall(L, 0, 0, 0)) {
+    let status;
+    {
```
Comment (giann) [inline tests/manual-tests/lua-cli.js]: why the block ?

File: tests/manual-tests/lua-cli.js [hunk 2]
```diff
@@ -24,10 +26,37 @@ for (;;) {
...
+        let buffer = lua.to_luastring(input);
+        status = lauxlib.luaL_loadbuffer(L, buffer, buffer.length, stdin);
+    }
+    if (status !== lua.thread_status.LUA_OK) {
+        lapi.lua_pop(L, 1);
+        let retbuffer = lua.to_luastring("return " + input);
```
Comment (giann) [inline tests/manual-tests/lua-cli.js]: What if input is already 'return ...' ?

File: tests/manual-tests/lua-cli.js (hunk 1 above)
Comment (daurnimator) [inline tests/manual-tests/lua-cli.js]: reduce scope of `buffer`.

File: tests/manual-tests/lua-cli.js (hunk 2 above)
Comment (daurnimator) [inline tests/manual-tests/lua-cli.js]: Then it was either valid syntax, or we try adding a second 'return' (same as what normal lua interpreter does)

Comment (giann) [inline tests/manual-tests/lua-cli.js]: Right: did not see it was if status !== LUA_OK

**Pull request merged**